server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchDocValuesContext.java

```diff
@@ -23,20 +24,23 @@
  */
 public class FetchDocValuesContext {
 
-    private final List<FieldAndFormat> fields = new ArrayList<>();
+    private final Collection<FieldAndFormat> fields;
 
     /**
      * Create a new FetchDocValuesContext using the provided input list.
      * Field patterns containing wildcards are resolved and unmapped fields are filtered out.
      */
     public FetchDocValuesContext(SearchExecutionContext searchExecutionContext, List<FieldAndFormat> fieldPatterns) {
+        // Use Linked HashMap to reserve the fetching order
+        final Map<String, FieldAndFormat> fieldToFormats = new LinkedHashMap<>();
         for (FieldAndFormat field : fieldPatterns) {
             Collection<String> fieldNames = searchExecutionContext.getMatchingFieldNames(field.field);
             for (String fieldName : fieldNames) {
-                fields.add(new FieldAndFormat(fieldName, field.format, field.includeUnmapped));
+                // the first matching field wins
+                fieldToFormats.putIfAbsent(fieldName, new FieldAndFormat(fieldName, field.format, field.includeUnmapped));
```

Comment:
I'm sorry, I remembered incorrectly what the `fields` option does. It always uses `put` on a `LinkedHashMap`, which means the last definition wins. I think it's worth doing the same for consistency/ simplicity.

Reply:
ok, I pushed https://github.com/elastic/elasticsearch/pull/89094/commits/e1342509564f7499510980736627923b33e53c18